bare_metal_tests/uart/test.py

```diff
@@ -0,0 +1,276 @@
+import logging
+import atexit
+
+"""! UART TEST FOR GREEN TEA
+
+  TEST SCENARIO
+  This part of the test is executed on the host (PC).
+  
+  Actions performed by the script:
+  - build mbed part of the UART test,
+  - request user to flash and reset the mbed board,
+  - perform uart communication test,
+  - report test status.
+  
+  Test scenario:
+  After startup script will try to communicate via UART with the mbed device.
+  This part of the test waits for the expected message and responds with
+  the same message. 
+  4 test messages should be received. Last message ("PASSED") informs
+  that on the mbed side also all messages have been received.
+ 
+  To perform this test with serial port baud rate different than default (9600),
+  call host test with appropriate --baudrate parameter.
+  
+  Example usage:
+      test.py -m K64F -t GCC_ARM -b 9600
+  This command will run test on K64F platfom using GCC_ARM toolchain and
+  serial connection speed equal to 9600 b/s.
+  
+  For more options type:
+      test.py -h
+"""
+
+RETRY_LIMIT = 10
+
+def serial_cleanup(serial_port):
+    """! Serial port cleanup handler.
+         This function closes opened serial port.
+         
+    @param serial_port serial port.
+    """
+    if (serial_port != None and serial_port.isOpen() == True):
+        serial_port.close()
+
+def run_command(cmd):
+    """! Runs command and prints proc stdout on screen.
+         It is used to build the test.
+         
+    @param cmd command to be executed.
+    @return image path.
+    """
+    image_path = str()
+    
+    # Remove build log file if exists
+    try:
+        os.remove('build.log')
+    except OSError:
+        pass
+    
+    try:
+        p = Popen(cmd, stdout=sys.stdout)
+    except OSError:
+        logging.exception ('Command can not be executed.')
+        sys.exit()
+
+    # Check process status.
+    returncode = p.wait()
+    if returncode != 0:
+        logging.error('Error while executing command.')
+        sys.exit()
+    
+    # Get image path   
+    try:
+        with open('build.log') as f:
+            build_data = json.load(f)
+            image_path = build_data['builds'][0]['bin']
+    # Image path can not be found.
+    except (KeyError, IndexError, IOError, AttributeError): 
+        # If we are here, then build process has failed.
+        logging.exception ('Unable to locate image file.')
+        sys.exit()
+       
+    return image_path
+
+def build_test(target, toolchain, clean_flag, baudrate):
+    """Build the mbed test.
+    
+    @param target target id.
+    @param toolchain compile toolchain.
+    @return image path.
+    """
+    logging.info('Building mbed UART test for green tea.')
+    
+    clean = ''
+    if (clean_flag):
+        clean = '-c'
+    
+    cmd = 'mbed compile -t %s -m %s --source . --source ../.. ' \
+          '--build-data build.log %s -D BAUD_RATE=%s' \
+          % (toolchain, target, clean, baudrate)
+    
+    return run_command(cmd)
```

Comment:
I can't see any benefit of `run_command` being a separate function. This might as well continue here, in `build_test`.